src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/VariantFlyweightGenerator.java

```diff
@@ -0,0 +1,1421 @@
+    private final MemberFieldGenerator memberField;
+    private final KindConstantGenerator memberKindConstant;
+    private final MemberSizeConstantGenerator memberSizeConstant;
+    private final MemberOffsetConstantGenerator memberOffsetConstant;
+    private final MemberFieldValueConstantGenerator memberFieldValueConstant;
+    private final TryWrapMethodGenerator tryWrapMethod;
+    private final WrapMethodGenerator wrapMethod;
+    private final ToStringMethodGenerator toStringMethod;
+    private final KindAccessorGenerator kindAccessor;
+    private final MemberAccessorGenerator memberAccessor;
+    private final LimitMethodGenerator limitMethod;
+    private final BuilderClassGenerator builderClass;
+    private static final Map<String, String> NUMBER_BY_WORD;
+    private GetMethodGenerator getMethod;
+    private BitMaskConstantGenerator bitMaskConstant;
+
+    static
+    {
+        Map<String, String> numberByWord = new HashMap<>();
+        numberByWord.put("0", "zero");
+        numberByWord.put("1", "one");
+        NUMBER_BY_WORD = unmodifiableMap(numberByWord);
+    }
+
+    public VariantFlyweightGenerator(
+        ClassName variantName,
+        ClassName flyweightName,
+        String baseName,
+        TypeName kindTypeName,
+        TypeName explicitTypeName,
+        TypeName unsignedExplicitTypeName)
+    {
+        super(variantName);
+
+        this.baseName = baseName;
+        this.builder = classBuilder(variantName).superclass(flyweightName).addModifiers(PUBLIC, FINAL);
+        this.memberField = new MemberFieldGenerator(variantName, kindTypeName, builder);
+        this.memberKindConstant = new KindConstantGenerator(variantName, kindTypeName, builder);
+        this.memberSizeConstant = new MemberSizeConstantGenerator(variantName, kindTypeName, builder);
+        this.memberOffsetConstant = new MemberOffsetConstantGenerator(variantName, kindTypeName, builder);
+        this.memberFieldValueConstant = new MemberFieldValueConstantGenerator(variantName, builder);
+        this.tryWrapMethod = new TryWrapMethodGenerator(kindTypeName);
+        this.wrapMethod = new WrapMethodGenerator(kindTypeName);
+        this.toStringMethod = new ToStringMethodGenerator(kindTypeName);
+        this.kindAccessor = new KindAccessorGenerator(variantName, kindTypeName, builder);
+        this.memberAccessor = new MemberAccessorGenerator(variantName, kindTypeName, builder);
+        this.limitMethod = new LimitMethodGenerator(kindTypeName);
+        this.builderClass = new BuilderClassGenerator(variantName, flyweightName, kindTypeName);
+        if (explicitTypeName != null)
+        {
+            this.getMethod = new GetMethodGenerator(kindTypeName, explicitTypeName, unsignedExplicitTypeName);
+            this.bitMaskConstant = new BitMaskConstantGenerator(variantName, explicitTypeName, builder);
+            builderClass.setExplicitType(explicitTypeName, unsignedExplicitTypeName);
+        }
```

Comment:
Just a thought, given that `explicitTypeName` is already inside the `getMethod` and `bitMaskConstant` generator implementations, we _could_ choose to always trigger these methods, but make them behave as a no-op when `explicitTypeName` is `null`.

In that case, all generator fields would be marked `final` and all the `if != null` can be removed.